eos/scattering/ee-to-ccbar_TEST.cc

```diff
@@ -0,0 +1,243 @@
+                auto D0Dbar0   = std::make_shared<PWavePPChannel<5, 2>>("D0Dbar0", p["mass::D^0"], p["mass::D^0"], p["ee->ccbar::q_0"], D0Dbar0_g0s);
+                auto DpDm      = std::make_shared<PWavePPChannel<5, 2>>("DpDm", p["mass::D^+"], p["mass::D^+"], p["ee->ccbar::q_0"], DpDm_g0s);
+                auto eff2S     = std::make_shared<EffChannel<5, 2>>("eff(2S)", p["ee->ccbar::effective_mass"], p["ee->ccbar::effective_mass"], p["ee->ccbar::q_0"], eff2S_g0s);
+                auto eff3770   = std::make_shared<EffChannel<5, 2>>("eff(3770)", p["ee->ccbar::effective_mass"], p["ee->ccbar::effective_mass"], p["ee->ccbar::q_0"], eff3770_g0s);
+
+                KMatrix<5, 2> KMatrix32({ee, D0Dbar0, DpDm, eff2S, eff3770}, {psi2S, psi3770}, bkgcst, "ee->ccbar");
+
+                auto tMatrix32s0 = KMatrix32.tmatrix_row(0,  0.0001);
+                auto tMatrix32s1 = KMatrix32.tmatrix_row(0, 13.94);
+                auto tMatrix32s2 = KMatrix32.tmatrix_row(0, 20.0);
+
+                TEST_CHECK_RELATIVE_ERROR(real(eff2S->chew_mandelstam(20)),   -0.0172797,  eps);
+                TEST_CHECK_RELATIVE_ERROR(imag(eff2S->chew_mandelstam(20)),    0.0189092,  eps);
+                TEST_CHECK_RELATIVE_ERROR(real(eff3770->chew_mandelstam(20)), -0.0172797,  eps);
+                TEST_CHECK_RELATIVE_ERROR(imag(eff3770->chew_mandelstam(20)),  0.0189092,  eps);
+
+                TEST_CHECK_RELATIVE_ERROR(tMatrix32s0[0].real(),  3.17782e-5,  eps);
+                TEST_CHECK_RELATIVE_ERROR(tMatrix32s0[0].imag(),  1.99853e-11, eps);
+                TEST_CHECK_RELATIVE_ERROR(tMatrix32s1[1].real(), -0.0391188,   eps);
+                TEST_CHECK_RELATIVE_ERROR(tMatrix32s1[1].imag(), -0.000556894, eps);
+                TEST_CHECK_RELATIVE_ERROR(tMatrix32s2[1].real(), -0.0067422,   eps);
+                TEST_CHECK_RELATIVE_ERROR(tMatrix32s2[1].imag(),  0.00187893,  eps);
+            }
+
+            {
+                // Test the full K matrix
+                Parameters p = Parameters::Defaults();
+                p["mass::psi(3770)"]                      =  3.796443282051135;
+                p["ee->ccbar::g0(psi(2S),e^+e^-)"]        = -0.02077753547690923;
+                p["ee->ccbar::g0(psi(3770),e^+e^-)"]      = -0.001999916489715092;
+                p["ee->ccbar::g0(psi(2S),D^0Dbar^0)"]     = -3.3693829070086214;
+                p["ee->ccbar::g0(psi(3770),D^0Dbar^0)"]   =  8.38428874933062;
+                p["ee->ccbar::g0(psi(2S),D^+D^-)"]        =  3.5692138012280807;
+                p["ee->ccbar::g0(psi(3770),D^+D^-)"]      = -8.391199026268827;
+                p["ee->ccbar::g0(psi(2S),eff(2S))"]       =  0.09237874214140328;
+                p["ee->ccbar::g0(psi(3770),eff(3770))"]   = -1.5409316160476978;
+
+                p["ee->ccbar::effective_mass"] = 0.1349768;
+
+                // Set the centrifugal factor to 0.5 GeV
```

Comment:
q_0 is not the centrifugal barrier factor, it is the effective momentum, isn't it?